### ISSUE TITLE
Webcam OS

### ISSUE THREAD

**Issue by photopea** [omnirom/android#58]
I don't know where to post it, I would like to get in touch with some people who understand Android OS development.

I think old Android phones could be reused as webcams. Cameras on old phones can be still better than cameras on new phones.

I would like someone to develop "Webcam OS", which can be deployed to any Android phone, and will turn it into a webcam. You could simply connect your phone to a laptop with a USB cable, and it will act as connecting any other USB webcam. It would not use any other inputs / outputs (no screen, no wifi, no buttons need to be pressed, etc). It could even work on phones with a broken / missing screens, and with no phone battery.

Are you aware of any such project? I think it could be useful to millions of people, as there are millions of people who have an old phone in their drawer. I am willing to sponsor such a project with several thousand USD.

**Comment by Entropy512** [comment]
Not really, because one of the biggest problems many Android BSP bringups have for "open source" people like Omni is lack of documentation for the camera sensor and image signal processor without an NDA.  You'll often find that the LAST thing that is working well in custom Android firmware for non-Pixel phones is the camera.

These would be just as big of a hurdle for anyone trying to deploy Buildroot/Yocto/etc on an Android device.

"no screen" = good luck flashing in the first place since nearly all Android device bootloaders display information on the screen.  u-boot serial consoles are virtually nonexistent in Android-land

"no battery" = 75%+ of Android phones have been designed with a hardware architecture that will cause them to refuse to boot at all if the battery is missing.

"several thousand USD" - At the salaries someone with the required skillset could pull in the commercial world, this is maybe 1-2 man-months worth of work.  Projects like Omni are full of volunteers that are involved for almost exclusively non-monetary reasons.  Common things are:

Personal skills development (This was one of the big reasons I worked on CyanogenMod and later Omni in my spare time - I was in a deadend job at Lockheed Martin I was deeply unhappy with and did Android to keep my brain from rotting.  Doing the work for any sort of monetary compensation would have been problematic in regards to my employment agreement with Lockmart.  One of the reasons I'm basically long retired from Android work is because I've had a new job since 2016 that gives me proper mental stimulation.)
Wanting to fix some sort of issue with your own personal phone - 90%+ of people who contribute to a project like CM or Omni are dissatisfied with some aspect of the stock firmware of their device and can't find any other device that meets their needs without modification.  This is the other reason I'm basically retired from Android development - stock Android has improved drastically over the years to the point where my dissatisfaction with the product is no longer sufficient to motivate me to spend time on changing it.  So the vast majority of my open source contributions are to photography nowadays.

**Comment by photopea** [comment]
I see, thank you very much for your comment! I was just curious, if it could be enough to delete or modify a few things in a regular Android release to create this.

I have another idea: when you connect your Android phone to a device nowadays, there are several "modes" (Charging, File transfer, USB tethering). Maybe there could be a new "webcam" mode, which would turn a phone into a webcam? So it would be a feature of Android itself.

Also, do you think it can be made as an Android app? To have a phone permanently connected to a computer, fully controled from a computer (so the app should never be killed by Android, and should "always listen" to the USB).
